### PR TITLE
feat: create a wallet without a provider

### PR DESCRIPTION
**Pull request by petertonysmith94**
closes https://github.com/FuelLabs/fuels-ts/issues/1444

### Issue

- Provider was made in the following [PR](https://github.com/FuelLabs/fuels-ts/pull/1181/files#diff-473114ce7ac2852f8838f96327e7171c9d8df3cb671e8f805d295b3603d0b1c4)
- User requested for the provider to be optional for Wallet instantion methods [here](https://github.com/FuelLabs/fuels-ts/issues/1444)

### Solution

- For wallet instantion, all methods with a provider argument should now be optional.
- Internally we set the provider as undefined.
- Any direct or indirect usage of the provider, will now throw an error if it's undefined.

### CODE REVIEW

**Comment by arboleya** [comment]
Following #1444 milestone, this PR should target the `rc/salamander` branch / [PR](https://github.com/FuelLabs/fuels-ts/pull/1495).

**Comment by danielbate** [comment]
@Torres-ssf it says you haven't signed the [CLA](https://github.com/FuelLabs/fuels-ts/pull/1566#issuecomment-1883547196) 🤯

**Comment by petertonysmith94** [comment]
Thank you @arboleya

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -43,6 +43,8 @@ jobs:
       - name: Find PR Number
         uses: jwalton/gh-find-current-pr@v1
         id: findPr
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (petertonysmith94) [inline .github/workflows/test.yaml:89]: Removing this now

File: .changeset/witty-beds-vanish.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
Comment (arboleya) [inline .changeset/witty-beds-vanish.md:2]: Changeset should not be empty.

**Review by arboleya — COMMENTED** [review]
This doesn't seem like a breaking change—_is it_?

The PR title incudes a `!`.

**Comment by petertonysmith94** [comment]
> This doesn't seem like a breaking change—_is it_?

No it doens't - removed the `!`

**Review by Torres-ssf — APPROVED** [review]
LGTM 🚀

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
@petertonysmith94, thank you for your valuable contribution to this project 🙏.

I've reviewed the changes and I think they're great. 

However, I'd like to suggest an enhancement for our documentation. Specifically, in the Wallet section, it would be beneficial to clarify that the provider is optional when creating Wallet/Account instances.

**Comment by petertonysmith94** [comment]
> @petertonysmith94, thank you for your valuable contribution to this project 🙏.
> 
> I've reviewed the changes and I think they're great.
> 
> However, I'd like to suggest an enhancement for our documentation. Specifically, in the Wallet section, it would be beneficial to clarify that the provider is optional when creating Wallet/Account instances.

You're most welcome - I have added a small point on Wallet creation with an example.
Please let me know if there is anything else 😃

**Review by Torres-ssf — APPROVED** [review]

**Review by danielbate — APPROVED** [review]
:shipit:

File: apps/docs/src/guide/wallets/access.md [hunk 3]
```diff
@@ -27,6 +27,12 @@ provides greater access via its private key.
 
 <<< @/../../../packages/fuel-gauge/src/doc-examples.test.ts#wallets{ts:line-numbers}
 
+## Optional Provider
+
+You can choose not to pass through a provider argument on `Wallet` construction:
+
+<<< @/../../../packages/fuel-gauge/src/doc-examples.test.ts#wallet-optional-provider{ts:line-numbers}
```
Comment (arboleya) [inline apps/docs/src/guide/wallets/access.md:34]: @FuelLabs/sdk-ts Just a reminder that code snippets should ideally come from the `apps/docs-snippets.`

**Pull request merged**